ocsmesh/raster.py

```diff
@@ -364,6 +364,19 @@ def __iter__(self, chunk_size: int = None, overlap: int = None):
         for window in self.iter_windows(chunk_size, overlap):
             yield window, self.get_window_bounds(window)
 
+    def __getstate__(self):
+        my_dict=super().__getstate__()
+        if 'source' in my_dict and hasattr(my_dict['source'], 'name'):
+            my_dict['source_path'] = my_dict['source'].name
+        my_dict.pop('source', None)
+        return my_dict
+
+    def __setstate__(self, state):
+        if 'source_path' in state:
+            state['source'] = rasterio.open(state['source_path'])
+            del state['source_path']
+        self.__dict__.update(state)
```

Comment:
We can probably do `super().__setstate__(state)` as well instead of the dict `update`, but I'm not sure